boostedblob/azure_auth.py

```diff
@@ -51,20 +51,13 @@ def load_credentials() -> Dict[str, Any]:
             "storage_account_key": connection_data["accountkey"],
         }
 
-    if "AZURE_APPLICATION_CREDENTIALS" in os.environ:
-        creds_path = os.environ["AZURE_APPLICATION_CREDENTIALS"]
-        if not os.path.exists(creds_path):
-            raise RuntimeError(
-                f"Credentials not found at '{creds_path}' specified by environment variable 'AZURE_APPLICATION_CREDENTIALS'"
-            )
-        with open(creds_path) as f:
-            creds = json.load(f)
-            return {
-                "_azure_auth": "svcact",
-                "client_id": creds["appId"],
-                "client_secret": creds["password"],
-                "tenant_id": creds["tenant"],
-            }
+    # MSI_ENDPOINT is what the Azure CLI uses to detect managed service identity
+    # https://github.com/Azure/azure-sdk-for-python/blob/2d61792d140ad895cfbf8f945454ca947cf638f2/sdk/identity/azure-identity/azure/identity/_constants.py#L47
+    if "MSI_ENDPOINT" in os.environ:
```

Comment:
I'm using `MSI_ENDPOINT` vs `AZURE_MSI_ENDPOINT` here because that's what's used in the azure docs and cli